example/fly_mission/fly_mission.cpp

```diff
@@ -0,0 +1,258 @@
+//
+// Example that demonstrates how add waypoint missions using DroneCore
+//
+// Author: Julian Oes <julian@oes.ch>, Shakthi Prashanth <shakthi.prashanth.m@intel.com>
+//
+
+#include <dronecore/dronecore.h>
+#include <iostream>
+#include <functional>
+#include <memory>
+#include <future>
+
+
+using namespace dronecore;
+using namespace std::placeholders; // for `_1`
+
+
+static std::shared_ptr<MissionItem> add_mission_item(double latitude_deg,
+                                                     double longitude_deg,
+                                                     float relative_altitude_m,
+                                                     float speed_m_s,
+                                                     bool is_fly_through,
+                                                     float gimbal_pitch_deg,
+                                                     float gimbal_yaw_deg,
+                                                     MissionItem::CameraAction camera_action);
+
+static void receive_mission_progress(int current, int total);
+
+static std::atomic<bool> _want_to_pause {false};
+
+int main(int /*argc*/, char ** /*argv*/)
+{
+    DroneCore dc;
+
+    {
+        auto prom = std::make_shared<std::promise<void>>();
```

Comment:
Should this promise include the connection error? We're using this to pass a UUID and I don't believe we handle the case where we get a timeout or no UUID.

Reply:
Arguably this could also be a synchronous fail case for the whole example - since we're never going to recover from not being able to connect. 
Unless we can presumably recover, in which case it would be nice to think about what sort of recovery options users have, and how they might implement them.

Reply:
Yes, we don't handle a timeout, we're just stuck forever. However, there won't ever be no UUID, and since we don't need the UUID outside, we don't need to include it in the promise.

Reply:
OK. As for other response, this makes sense - was just not what I expected based on previous example.